mypy/constraints.py

```diff
@@ -764,16 +774,22 @@ def visit_instance(self, template: Instance) -> list[Constraint]:
                         if isinstance(suffix, (Parameters, CallableType)):
                             # no such thing as variance for ParamSpecs
                             # TODO: is there a case I am missing?
-                            # TODO: constraints between prefixes
-                            prefix = template_arg.prefix
 
                             suffix = suffix.copy_modified(
-                                suffix.arg_types[len(prefix.arg_types) :],
-                                suffix.arg_kinds[len(prefix.arg_kinds) :],
-                                suffix.arg_names[len(prefix.arg_names) :],
+                                suffix.arg_types[length:],
+                                suffix.arg_kinds[length:],
+                                suffix.arg_names[length:],
                             )
                             res.append(Constraint(template_arg, SUPERTYPE_OF, suffix))
                         elif isinstance(suffix, ParamSpecType):
+                            suffix_prefix = suffix.prefix
+                            constrained = suffix.copy_modified(
```

Comment:
constrained isn't used here

Reply:
I think it should be. Uh, I can see that we're missing a bunch of test cases... and I'm not exactly sure how to even hit this code! I'll try adding some more descriptive comments of my thought process at least.